packages/rum-core/src/domain/foregroundContexts.ts

```diff
@@ -12,8 +12,10 @@ import {
 } from '@datadog/browser-core'
 import { InForegroundPeriod } from '../rawRumEvent.types'
 
-// Arbitrary value to cap number of element (mostly for backend)
-export const MAX_NUMBER_OF_FOCUSED_TIME = 500
+// Arbitrary value to cap number of element mostly for backend & to save bandwidth
+export const MAX_NUMBER_OF_FOCUSED_TIME_PER_VIEW = 500
+// Arbitrary value to cap number of element mostly for memory consumption in the browser
+export const MAX_NUMBER_OF_FOCUSED_TIME = 2500
```

Comment:
Do we need to keep and increase `MAX_NUMBER_OF_FOCUSED_TIME`?

Reply:
The idea is to have  two  limits  serving two different purposes;

* `MAX_NUMBER_OF_FOCUSED_TIME` is here to limit the memory usage. It can be fairly large as objects are small.
* `MAX_NUMBER_OF_FOCUSED_TIME_PER_VIEW` is here to limit  the bandwidth usage. We keep the same limit as before (500) as it should be enough.

Reply:
Agreed. Most of the views have less than 500 focused periods therefore the array is full because of SPA changes. We can drastically reduce the number of errors while keeping the same limit for the bandwidth.

Reply:
IIUC:
- `MAX_NUMBER_OF_FOCUSED_TIME` is the max number of foreground periods stored 
- `MAX_NUMBER_OF_FOCUSED_TIME_PER_VIEW` is the max number of foreground periods than can be returned by `getInForegroundPeriods`.

if that so, we could probably clarify the namings:
- using "view" in the constant is "leaking" information about the current consumer which we don't need to know here
- I am confused by "focused time" vs "foreground period", is there a difference?
- we could maybe be more expressive than `get` for `getInForeground` and `getInForegroundPeriods` and use the same verb for `getInForegroundPeriods` and the related constant.

Given that, I would have in mind something like:

```ts
const MAX_NUMBER_OF_SELECTABLE_FOREGROUND_PERIODS = 500
const MAX_NUMBER_OF_STORED_FOREGROUND_PERIODS = 2500

isInForegroundAt: (startTime: RelativeTime) => boolean | undefined
selectInForegroundPeriodsFor: (startTime: RelativeTime, duration: Duration) => InForegroundPeriod[] | undefined
```

wdyt?

Reply:
Agreed, the naming you're proposing is better than the current one